solana/solana-ibc/programs/solana-ibc/src/chain.rs

```diff
@@ -150,6 +155,27 @@ impl ChainData {
             .map_err(into_error)
     }
 
+    /// Returns the validator data with stake and rewards
+    pub fn get_validator(
+        &self,
+        validator: Pubkey,
+    ) -> Result<Option<Validator>, ChainNotInitialised> {
+        let inner = self.get()?;
+        Ok(inner
+            .manager
+            .get_validators()
+            .iter()
+            .find(|&c| c.pubkey == validator.into())
+            .cloned())
+    }
+
+    pub fn get_staking_program_id(
+        &self,
+    ) -> Result<Pubkey, ChainNotInitialised> {
+        let chain_inner = self.get()?;
+        Ok(*chain_inner.staking_program_id)
+    }
+
```

Comment:
Firstly, as before, Rustitc way to name this would be to not use
`get_` prefix.

Secondly, alternative would be to have a `check_staking_program`
method instead:

```suggestion
    /// Checks whether given `program_id` matches expected staking program id.
    ///
    /// The staking program id is stored within the chain account.  Various
    /// CPI calls which affect stake and rewards can only be made from that
    /// program.  This method checks whether program id given as argument
    /// matches the one we expect.  If it doesn’t, returns `InvalidCPICall`.
    pub fn check_staking_program(
        &self,
        program_id: &Pubkey
    ) -> Result<(), Error> {
        match program_id == &self.get()?.staking_program_id {
            false => Err(Error::InvalidCPICall),
            true => Ok(()),
        }
    }

```

Dunno if this is necessarily better.  Up to you.

Reply:
lets go with the above method.